apigw-custom-domain/Makefile

```diff
@@ -2,15 +2,28 @@ export AWS_ACCESS_KEY_ID ?= test
 export AWS_SECRET_ACCESS_KEY ?= test
 export AWS_DEFAULT_REGION ?= us-east-1
 
-
 usage:       ## Show this help
 	@fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##//'
 
+precheck:    ## Check if all required prerequisites are installed
+	@command -v docker > /dev/null 2>&1 || { echo "Docker is not installed. Please install Docker and try again."; exit 1; }
+	@command -v node > /dev/null 2>&1 || { echo "Node.js is not installed. Please install Node.js and try again."; exit 1; }
+	@command -v aws > /dev/null 2>&1 || { echo "AWS CLI is not installed. Please install AWS CLI and try again."; exit 1; }
+	@command -v python > /dev/null 2>&1 || { echo "Python is not installed. Please install Python and try again."; exit 1; }
+	@command -v openssl > /dev/null 2>&1 || { echo "OpenSSL is not installed. Please install OpenSSL and try again."; exit 1; }
+	@echo "All required prerequisites are available."
+
 install:     ## Install dependencies
-	@npm install
 	@which serverless || npm install -g serverless
 	@which localstack || pip install localstack
 	@which awslocal || pip install awscli-local
+	@if [ ! -d "node_modules" ]; then \
+		echo "node_modules not found. Running npm install..."; \
+		npm install; \
+	else \
+		echo "node_modules already installed."; \
+	fi
+	@echo "All required dependencies are available."
 
 cert:        ## Create test SSL certificate
```

Comment:
Should this step be carried over a shell script?

Reply:
I think this is okay as is, unless you want to move the whole cert generation to run.sh, as it's really part of the deployment, and not a pre-requisite